streams-for-raw-video-explainer.md

```diff
@@ -0,0 +1,124 @@
+# Transfering Ownership Streams Explained
+
+
+## Introduction
+
+The streams APIs provide a convenient way to build processing pipelines.
+Using streams with `VideoFrame` objects has known shortcomings, as illustrated by https://github.com/whatwg/streams/issues/1155 or https://github.com/whatwg/streams/issues/1185 for instance.
+This proposal addresses these shortcomings by improving streams support for objects similar but not limited to WebCodec `VideoFrame`.
+It could also be useful for streams making use of `ArrayBuffer` or objects that own `ArrayBuffer` like `RTCEncodedVideoChunk`.
+
+Streams APIs can create coupling between the processing units of the pipeline when objects in the pipe are mutable:
+a processing unit A might pass an object O to another unit B through a stream (say a `WritableStream`) W.
+If A keeps a reference to O, it might mutate O while B is doing processing based on O.
+
+This is especially an issue with objects like `VideoFrame` objects that need to be closed explicitly.
+Taking the previous example, if A decides to close a `VideoFrame` after passing it to W but before B gets it, B will receive a closed `VideoFrame`, which is probably considered a bug.
+If A does not close `VideoFrame`, it is the responsibility of the remaining of the pipeline to close it.
+There is a need to clearly identify who is owning these objects and who is responsible to close these objects at any point in time.
+
+The proposed solution is to transfer ownership of an object to the stream when it gets written or enqueueud to the stream.
+By doing so, the processing unit that enqueues/writes objects will not be able to mutate objects manipulated by the stream and is relieved of the lifetime management of these objects.
+Conversely, processing units take ownership of objects when they receive them from a stream.
+
+Transferring ownership should be opt-in. For that purpose, a new streams type, named 'transfer' in this document,  would be added.
+
+## Example
+
+Below is an example of JavaScript that shows how this can be used.
+The example creates a processing pipe starting with a camera stream and applying two transforms, one for doing a processing for every 30 frame, and one for doing background blur.
+
+```javascript
+// JS transform
```

Comment:
I think it's important we lead with worker examples from the get-go.
```suggestion
// worker.js
```

Reply:
Right